srcs/parsing/Parser.cpp

```diff
@@ -0,0 +1,183 @@
+
+// Copy constructor
+Parser::Parser(const Parser &other): rh(other.rh), _server(other._server), _client(other._client) {}
+
+// Assignment operator overload
+Parser &Parser::operator=(const Parser &other)
+{
+	if (this != &other) {
+		rh = other.rh;
+		_client = other._client;
+		_server = other._server;
+	}
+    return (*this);
+}
+
+// Destructor
+Parser::~Parser(void) {}
+
+bool Parser::response(ReplyCode code, const std::string& params, const std::string& trailing) {
+	if (code != CORRECT_FORMAT) {
+		if (rh && _client) {
+			rh->process_response(*_client, code, params, NULL, trailing);
+		}
+		return (false);
+	}
+	return (true);
+}
+
+bool Parser::correct_password(std::string& password)
+{
+
+	std::cout << "CORRECT PASSWORD" << std::endl;
+    if (_server->get_password() != password) {
+        return response(ERR_PASSWDMISMATCH);
+    } else if (_client->get_status() == AUTHENTICATED) {
+		std::cout << "AUTHENTICATED" << std::endl;
+        return response(ERR_ALREADYREGISTRED);
+    } else if (_client->get_status() == REGISTERED) {
+		std::cout << "REGISTERED" << std::endl;
+	}
+	std::cout << "TRUE" << std::endl;
+    return (true);
+}
+
+bool Parser::correct_channel(std::string& name)
+{
+    if (name.empty()) {
+		return (response(ERR_BADCHANMASK));
+    }
+    for (std::string::const_iterator it = name.begin(); it != name.end(); ++it) {
+        unsigned char c = *it;
+        if (c > 0xFF || Utils::is_char_of(c, std::string(FORBIDEN_CHAR_CHAN_NAME, 7))) { // NOLINT
+			return response(ERR_BADCHANMASK, name);
+        }
+    }
+    if (!Utils::is_char_of(static_cast<unsigned char>(name[0]), "#&+!")) {
+		return response(ERR_BADCHANMASK, name);
+	}
+	if (name.length() <= 1 || name.length() >= ircConfig.get_chan_name_max_len()) {
+		return response(ERR_BADCHANMASK, name);
+	}
+
+	return true;
+}
+
+// not used
+bool Parser::correct_target(std::string& target)
+{
+	Parser parser;
+
+	bool validChannel = parser.correct_channel(target);
+	bool validNickname = parser.correct_nickname(target);
+
+	if (!validChannel && !validNickname)
+		return response(ERR_NOTONCHANNEL);
+
+	return CORRECT_FORMAT;
+}
+
+/**
+ * @brief check if the key is valid -- no commas or spaces allowed
+ *
+ * @param key
+ * @return true or false
+ */
+bool Parser::correct_key(std::string& key)
+{
+    if (key.empty()) {
+        return true;
+    } else if (key.size() > CHAN_KEY_MAX_LEN) {
+        return response(ERR_BADCHANNELKEY);
+    }
+    for (std::string::const_iterator it = key.begin(); it != key.end(); ++it) {
+        unsigned char c = *it;
+        if (c > 0x07F || Utils::is_char_of(c, std::string(FORBIDEN_CHAR_CHAN_KEY, 7))) { // NOLINT
+            return response(ERR_BADCHANNELKEY);
+        }
+    }
+    return true;
+}
+
+bool Parser::correct_nickname(std::string& nickname)
+{
+    bool 			invalidChar = std::count_if(nickname.begin(), nickname.end(), Utils::is_invalid_char_nick);
```

Comment:
the clang tidy check passes if we initialize it like this
```suggestion
    bool 			invalidChar = false;
```